### PR TITLE
signing: add signer with genesis id

### PR DESCRIPTION
**Pull request by evt**
## Motivation

Closes #[3503](https://github.com/spacemeshos/go-spacemesh/issues/3503)

## Changes

Add signer with genesis id, add test

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: activation/activation_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func (n *NetMock) hookToAtxPool(transmission []byte) {
 }
 
 func NewMockSigner() *MockSigning {
-	return &MockSigning{signing.NewEdSigner()}
+	return &MockSigning{signing.NewEdSigner([20]byte{})}
```
Comment (dshulyak) [inline activation/activation_test.go:101]: i think this is why using config option was a better approach

File: signing/signer.go [hunk 2]
```diff
@@ -154,23 +163,24 @@ func Verify(pubkey *PublicKey, message []byte, sign []byte) bool {
...
-
 // Verify that signature matches public key.
-func (EDVerifier) Verify(pub *PublicKey, msg, sig []byte) bool {
+func (ev EdVerifier) Verify(pub *PublicKey, msg, sig []byte) bool {
+	if len(msg) < 20 {
+		return false
+	}
+	if string(msg[0:20][:]) != string(ev.genesisID[:]) {
+		return false
+	}
```
Comment (dshulyak) [inline signing/signer.go:178]: this implementation is incorrect. message should not include genesis id. verifier should prefix message before attempting to extract public key

**Comment by evt** [comment]
Closed in favour of [3597](https://github.com/spacemeshos/go-spacemesh/pull/3597)

File: activation/activation_test.go (hunk 1 above)
Comment (evt) [reply activation/activation_test.go:101]: Made it as an option in https://github.com/spacemeshos/go-spacemesh/pull/3597

File: signing/signer.go (hunk 2 above)
Comment (evt) [reply signing/signer.go:178]: done this way in https://github.com/spacemeshos/go-spacemesh/pull/3597

**Pull request closed without merging**